File: code.c
/*
TEJ3M-Unit #2-07 arduino
created by Sophie Nguyen 
created on Apr 1

servo turns depending on distance from sonar
*/

#include <Servo.h>

Servo servoNumber1;

//constants
const int TRIG_PIN = 3;  
const int ECHO_PIN = 2; 
const float SPEED_OF_LIGHT = 0.0343;

//variables
float duration, distance;  

//setup
void setup() {  
   // setup servo pins
   servoNumber1.attach(4);
   servoNumber1.write(0);
  //pins for sonar
   pinMode(TRIG_PIN, OUTPUT);  
   pinMode(ECHO_PIN, INPUT); 
   Serial.begin(9600);  
}  


//loop
void loop() {  
  //clears the trig pin
  digitalWrite(TRIG_PIN, LOW);  
  delayMicroseconds(2);  

  //sets trig pin on high fpr 10 micro seconds
  digitalWrite(TRIG_PIN, HIGH);  
  delayMicroseconds(10);  
  digitalWrite(TRIG_PIN, LOW);  

  //reads the echo pin, returns the sound wave travel time in microseconds
  duration = pulseIn(ECHO_PIN, HIGH);  

  distance = duration*SPEED_OF_LIGHT/2;
  Serial.print(distance);
  
  if (distance < 50){
    servoNumber1.write(90);
    delay(500);
    servoNumber1.write(0);
    delay(500);
  }

}

File: code.py
"""
Created by: Sophie N
on: Apr 16

servo turns depending on distance from sonar
"""

import time
import board
import adafruit_hcsr04
import digitalio
import pwmio
import servo


# variables
distance = 0
servo_delay = 2
TOO_CLOSE = 20

# setup
sonar = adafruit_hcsr04.HCSR04(trigger_pin = board.GP15, echo_pin = board.GP14)

# create a PWMOut object on Pin GP12.
pwm = pwmio.PWMOut(board.GP12 , duty_cycle=2 ** 15, frequency=50)

# Create a servo object, my_servo.
my_servo = servo.Servo(pwm)

# loop
while True:
    # Sonar gets the distance form object
    distance = sonar.distance

    # Turns on LED if an object’s distance is equal to or closer then 20 cm from the sonar
    if distance < TOO_CLOSE:
            my_servo.angle = 180
    else:
            my_servo.angle = 0

    time.sleep(servo_delay)